src/ObjectPromiseProxy.js

```diff
@@ -56,44 +35,82 @@ function ObjectPromiseProxy (promise, target) {
           // 500 doesn't return a parsable response
         }
         // TODO: add all errors from the API response to the target
-        target.errors = {
-          ...target.errors,
+        // also TODO: split server errors by target once the info is available from the API
+        // maybe handle not just the first target lol
+        targets[0].errors = {
+          ...targets[0].errors,
           status: status,
           base: [{ message: message }],
           server: json.errors
         }
 
-        const errorString = JSON.stringify(target.errors)
+        const errorString = JSON.stringify(targets[0].errors)
         return Promise.reject(new Error(errorString))
       }
     },
     function (error) {
-      // TODO: Handle error states correctly
-      target.isInFlight = false
+      // TODO: Handle error states correctly, including handling errors for multiple targets
+      targets.forEach(t => { t.isInFlight = false })
       target.errors = error
       throw error
       // return target
     }
   )
   // Define proxied attributes
-  const attributeNames = Object.keys(target.attributeNames)
+  const attributeNames = Object.keys(targets[0].attributeNames)
   const tempProperties = attributeNames.reduce((attrs, key) => {
     attrs[key] = {
-      value: target[key],
+      value: targets[0][key],
       writable: false
     }
     return attrs
   }, {})
 
+  // TODO: how is isInFlight used? how should it be set for multiple targets?
   Object.defineProperties(result, {
-    isInFlight: { value: target.isInFlight },
+    isInFlight: { value: targets[0].isInFlight },
     ...tempProperties
   })
 
   // Return promise
   return result
 }
 
+// TODO: can this be merged with createOrUpdateModel?
```

Comment:
Quite possibly! One thing to be aware of if you do, though: We need to be able to distinguish between these two scenarios:
* creating a new model instance client side (which has _not_ been persisted by the API)
* creating a model from an API response (which ostensibly _has_ been persisted by the API)

The distinction is manifest when creating the initial snapshot:

```javascript
target._takeSnapshot({ persisted: true })
```

Now the `Model` constructor does include this:
```javascript
this._takeSnapshot({ persisted: !this.isNew })
```

`isNew` checks for a (non-temporary) `id`, so maybe we don't even need to be taking snapshots in this context at all, since the `Model` constructor should handle both cases (assuming responses from the server have an `id` and "ephemeral" instances do not. We would need to investigate a little bit more and verify though. 👍